apm-sniffer/apm-toolkit-activation/apm-toolkit-logback-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/logback/v1/x/PrintTraceIdInterceptor.java

```diff
@@ -37,7 +38,11 @@
 
     @Override public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
         Class<?>[] argumentsTypes, Object ret) throws Throwable {
-        return "TID:" + ContextManager.getGlobalTraceId();
+        String tid = ContextManager.getGlobalTraceId();
+        if ("N/A".equals(tid)) {
+            tid = (String) ((EnhancedInstance) allArguments[0]).getSkyWalkingDynamicField();
+        }
```

Comment:
Why couldn't getDynamicField first? Then call `ContextManager#getGlobalTraceId`, should be more efficient.

Reply:
OK